src/apply_rules/object_rules.rs

```diff
@@ -0,0 +1,508 @@
+        CommercialMode, Line, Network, ObjectType as ModelObjectType, PhysicalMode, VehicleJourney,
+    },
+    report::{Report, TransitModelReportCategory},
+    Result,
+};
+use failure::format_err;
+use log::info;
+use relational_types::IdxSet;
+use serde::Deserialize;
+use serde_json::{error::Category, error::Error as SerdeJsonError, Value};
+use std::{
+    collections::{HashMap, HashSet},
+    fs::File,
+    path::Path,
+    result::Result as StdResult,
+};
+
+#[derive(Debug, Deserialize)]
+struct ObjectProperties {
+    #[serde(flatten)]
+    properties: HashMap<String, Value>,
+    #[serde(default)]
+    grouped_from: Vec<String>,
+}
+
+fn get_value_string_from_properties(
+    property: &serde_json::value::Value,
+    key: &str,
+    default_value: &str,
+) -> String {
+    if let Some(value) = property.get(key) {
+        String::deserialize(value)
+            .map(|opt| opt)
+            .unwrap_or_else(|_| default_value.to_string())
+    } else {
+        default_value.to_string()
+    }
+}
+
+fn get_opt_value_string_from_properties(
+    property: &serde_json::value::Value,
+    key: &str,
+    default_value: Option<String>,
+) -> Option<String> {
+    if let Some(value) = property.get(key) {
+        Option::<String>::deserialize(value)
+            .map(|opt| opt)
+            .unwrap_or(default_value)
+    } else {
+        default_value
+    }
+}
+
+fn get_opt_value_number_from_properties(
+    property: &serde_json::value::Value,
+    key: &str,
+    default_value: Option<u32>,
+) -> Option<u32> {
+    if let Some(value) = property.get(key) {
+        Option::<u32>::deserialize(value)
+            .map(|opt| opt)
+            .unwrap_or(default_value)
+    } else {
+        default_value
+    }
+}
+
+fn check_and_apply_physical_modes_rules(
+    report: &mut Report<TransitModelReportCategory>,
+    mut collections: Collections,
+    physical_modes_rules: Vec<ObjectProperties>,
+    vjs_by_physical_mode: &HashMap<String, IdxSet<VehicleJourney>>,
+) -> Result<Collections> {
+    info!("Checking physical modes rules.");
+    let mut physical_modes_to_remove: HashSet<String> = HashSet::new();
+    let mut new_physical_modes: Vec<PhysicalMode> = vec![];
+
+    for pyr in physical_modes_rules.into_iter() {
+        let properties = pyr
+            .properties
+            .get("properties")
+            .ok_or_else(|| format_err!("Object \"properties\" is required"))?;
+
+        let physical_mode_id = properties
+            .get("physical_mode_id")
+            .ok_or_else(|| format_err!("Key \"physical_mode_id is required"))?
+            .as_str()
+            .unwrap();
+
+        if let Some(mut physical_mode) = collections.physical_modes.get_mut(physical_mode_id) {
+            physical_mode.name = get_value_string_from_properties(
+                properties,
+                "physical_mode_name",
+                &physical_mode.name,
+            );
+        } else if !CO2_EMISSIONS.contains_key(physical_mode_id) {
```

Comment:
Careful, `CO2_EMISSIONS` doesn't contains all the possible Physical Modes listed in the NTFS specification.